### PR TITLE
Adding support for Bootstrap panels

### PR DESCRIPTION
**Pull request by heryqk**
http://getbootstrap.com/components/#panels

Added mods requested by @martin-g

### CODE REVIEW

File: bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/panel/BootstrapGenericPanel.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+		Panel panelFooter = newFooterPanel(_PANEL_FOOTER_ID, getModel());
+		add(panelFooter);
+		Components.hideIfModelIsEmpty(panelFooter);
+		
+	}
+	
+	@Override
+	protected void onBeforeRender() {
+	
+		add(new PanelBehavior(getPanelType()));
```
Comment (martin-g) [inline bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/panel/BootstrapGenericPanel.java]: What is the reason to re-add a new instance of the behavior on each render ?

**Comment by martin-g** [comment]
Thanks!
Looks good! Just one question above about the re-adding of the PanelBehavior on each render of the panel.

File: bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/panel/BootstrapGenericPanel.java (hunk 1 above)
Comment (heryqk) [inline bootstrap-core/src/main/java/de/agilecoders/wicket/core/markup/html/bootstrap/panel/BootstrapGenericPanel.java]: It was a failed attempt at allowing the PanelType to be changed at runtime through an Ajax call for example, where through the click of a button, one could change the PanelType. 

I have made modifications. Submitting PR.

Thanks for your feedback.

**Comment by martin-g** [comment]
Thank you!

**Pull request merged**